portal-web/docroot/html/js/liferay/ddm_form.js

```diff
@@ -282,6 +282,31 @@ AUI.add(
 						);
 					},
 
+					_addFieldValidation: function(newField, originalField) {
+						var instance = this;
+
+						instance.fire(
+							'repeat',
+							{
+								field: newField,
+								originalField: originalField
+							}
+						);
+
+						AArray.each(
+							newField.get('fields'),
+							function(item, index) {
+								var name = item.get('name');
+
+								var originalChildFields = originalField.getFieldsByName(name);
+
+								if (originalChildFields.length) {
+									instance._addFieldValidation(item, originalChildFields[0]);
```

Comment:
Yes. That is correct.

I took into consideration that I didn't need the whole array and was going to use `.find()` instead, but didn't feel I would get any performance improvement, as the DOM was already queried at that point.

In `ee-6.2.x` there was a similar method called `getFieldsList`, now `getFieldNodes`, which you could pass in an optional `fieldName`. The new `getFieldNodes` removed the `fieldName` option.

I was replicating this "named" functionality. Rather than create a single use method, I wanted to create a more general one that someone else could use. I now realize this is a case of YAGNI (You aren't gonna need it).

I'm fine changing it to `.find()`.